raster/src/main/scala/geotrellis/raster/CellTypeEncoding.scala

```diff
@@ -24,40 +26,82 @@ import scala.util.Try
-  case object int16 extends FixedNoDataEncoding
-  case object uint16 extends FixedNoDataEncoding
-  case object int32 extends FixedNoDataEncoding
-  case object float32 extends FixedNoDataEncoding
-  case object float64 extends FixedNoDataEncoding
-  case object int8ud extends UserDefinedNoDataEncoding
-  case object uint8ud extends UserDefinedNoDataEncoding
-  case object int16ud extends UserDefinedNoDataEncoding
-  case object uint16ud extends UserDefinedNoDataEncoding
-  case object int32ud extends UserDefinedNoDataEncoding
+  sealed trait CellTypeEncoding {
+    val name: String
+  }
+  case object bool extends FixedNoDataEncoding {
+    override val name = "bool"
+  }
+  case object boolraw extends FixedNoDataEncoding {
+    override val name = "boolraw"
+  }
+  case object int8raw extends FixedNoDataEncoding {
+    override val name = "int8raw"
+  }
+  case object uint8raw extends FixedNoDataEncoding {
+    override val name = "uint8raw"
+  }
+  case object int16raw extends FixedNoDataEncoding {
+    override val name = "int16raw"
+  }
+  case object uint16raw extends FixedNoDataEncoding {
+    override val name = "uint16raw"
+  }
+  case object int32raw extends FixedNoDataEncoding {
+    override val name = "int32raw"
+  }
+  case object float32raw extends FixedNoDataEncoding {
+    override val name = "float32raw"
+  }
+  case object float64raw extends FixedNoDataEncoding {
+    override val name = "float64raw"
+  }
+  case object int8 extends FixedNoDataEncoding {
+    override val name = "int8"
+  }
+  case object uint8 extends FixedNoDataEncoding {
+    override val name = "uint8"
+  }
+  case object int16 extends FixedNoDataEncoding {
+    override val name = "int16"
+  }
+  case object uint16 extends FixedNoDataEncoding {
+    override val name = "uint16"
+  }
+  case object int32 extends FixedNoDataEncoding {
+    override val name = "int32"
+  }
+  case object float32 extends FixedNoDataEncoding {
+    override val name = "float32"
+  }
+  case object float64 extends FixedNoDataEncoding {
+    override val name = "float64"
+  }
+  case object int8ud extends UserDefinedNoDataEncoding {
+    override val name = "int8ud"
+  }
+  case object uint8ud extends UserDefinedNoDataEncoding {
+    override val name = "uint8ud"
+  }
+  case object int16ud extends UserDefinedNoDataEncoding {
+    override val name = "int16ud"
+  }
+  case object uint16ud extends UserDefinedNoDataEncoding {
+    override val name = "uint16ud"
+  }
+  case object int32ud extends UserDefinedNoDataEncoding {
+    override val name = "int32ud"
+  }
   case object float32ud extends UserDefinedNoDataEncoding {
+    override val name = "float32ud"
     override val isFloatingPoint = true
   }
   object float64ud extends UserDefinedNoDataEncoding {
+    override val name = "float64ud"
     override val isFloatingPoint = true
   }
 }
 
-/** Root trait for CellType encoding definitions. */
-sealed trait CellTypeEncoding {
-  def name = getClass.getName.split('$').last
```

Comment:
marking this as `lazy val` didn't help?

Reply:
I didn't bother.... for some reason I went the brute force route.... maybe because that creates a `synchronized` block.

Feel strongly it should be that way instead?

Reply:
Nope, just thinking in a way of minimizing changes and hand code writing. I'll check my idea in a spare time. Could you please add into the issue the way we can observe the memory pressure? Probably lots of people would be interested in it, and thanks for a quick PR 💯 !

Reply:
From https://github.com/locationtech/geotrellis/issues/3070:

> In a large compute job it became evident that something was awry with memory allocations. During the RasterFrames process of decoding `Tile`s from catalyst memory, the `CellType` representation is converted from a `String` into a `case object`. That process involves comparing names, etc. The `CellType` `name` method actually constructs the name from it's `case object` class name.... but doesn't cache the value.

<img width="1175" alt="Screen Shot 2019-09-12 at 12 24 41 PM" src="https://user-images.githubusercontent.com/131013/64802430-a3980c80-d558-11e9-9ae8-8631c06c73b1.png">